Import-VIRole.ps1

```diff
@@ -88,11 +87,19 @@ function Import-VIRole
         $null = Set-PowerCLIConfiguration -InvalidCertificateAction Ignore -DisplayDeprecationWarnings:$false -Scope User -Confirm:$false
 
         Write-Verbose -Message "Connecting to vCenter server '$vCenter'"
+        
         Try 
         {
-            $null = Connect-VIServer -Server $vCenter -ErrorAction Stop -Session ($global:DefaultVIServers | Where-Object -FilterScript {
+            if (($global:defaultViServer -eq $vCenter) -and (Get-VIRole)) {
+              #Already Have a Valid connection
+              $fred = Get-VIRole
```

Comment:
What is `$fred` used for? Since I do not see it elsewhere, would `$null` work? Or, if you prefer something to indicate it holds an existing session, maybe `$existingSession`?